app/controllers/bids_controller.rb

```diff
@@ -61,4 +63,22 @@ def is_owner_of_bid?
       redirect_to root_path
     end
   end
+
+  def ensure_less_then_ten_bids?
+    @bids = Bid.where(bidder_id: current_user.id, bidder_type: current_user.class.name)
+    @bids_this_month = @bids.this_month.count
+    if @bids_this_month >= 10 && current_user.account_type == 0 && broker_signed_in?
+      flash[:alert] = "Sorry! You can't bid on more then 10 jobs please switch to premium plan."
+      redirect_to root_path
+    end
+  end
+
+  def ensure_less_then_fifteen_bids?
+    @bids = Bid.where(bidder_id: current_user.id, bidder_type: current_user.class.name)
+    @bids_this_month = @bids.this_month.count
+    if @bids_this_month >= 1 && current_user.account_type == 0 && forward_freight_signed_in?
+      flash[:alert] = "Sorry! You can't bid on more then 15 jobs please switch to premium plan."
+      redirect_to root_path
+    end
+  end
```

Comment:
Hey @rabingaire, this is a good start, but there are a few things we could improve:

- This doesn't ensure that more than ten/fifteen bids are created at the model level, so it's fragile (model validations would be better)
- We can encapsulate this logic into the user model instead

```
# broker.rb
def bid_limit_exceeded?
  return if account_type == 0 
  bids.this_month.count >= 10
end

# forward_freight.rb
def bid_limit_exceeded?
  return if account_type == 0 
  bids.this_month.count >= 15
end
```

Then call this in the controller before_action like such:

```
def ensure_bid_limit
  return unless current_user.bid_limit_exceeded?
  flash[:alert] = "Sorry! You can't bid on more then 15 jobs please switch to premium plan."
  redirect_to root_path
end
```

The great thing about doing this is that we can create a method with the same name in broker and ff user models, and we can call it on `current_user`.

By doing this, we can eliminate two methods (`ensure_less_then_ten_bids?` and `ensure_less_then_fifteen_bids?`) and make it into one method called `ensure_bid_limit`

Pretty cool right 😄